packages/sync/src/Replicastore.php

```diff
@@ -1202,6 +1202,41 @@ public function get_checksum_columns_for_object_type( $object_type ) {
 		}
 	}
 
+	/**
+	 * Grabs the minimum and maximum object ids for the given parameters.
+	 *
+	 * @access public
+	 *
+	 * @param string $id_field     The id column in the table to query.
+	 * @param string $object_table The table to query.
+	 * @param string $where        A sql where clause without 'WHERE'.
+	 * @param int    $bucket_size  The maximum amount of objects to include in the query.
+	 *                             For `term_relationships` table, the bucket size will refer to the amount
+	 *                             of distinct object ids. This will likely include more database rows than
+	 *                             the bucket size implies.
+	 *
+	 * @return object An object with min_id and max_id properties.
+	 */
+	public function get_min_max_object_id( $id_field, $object_table, $where, $bucket_size ) {
```

Comment:
I've been thinking more about this method and realized that this doesn't make sense to be in the replicastore at all.

Currently, just because we placed this method in the replicastore, we need to pass it an `$id_field` and `$object_table`. 

Instead, if we put this method in the `Module` class, we'll be able to infer the ID field and the table name from the module object.

Another good reason to move this method there is the fact that it is called only from the `Module` class; so even only at an abstraction level it would make more sense there.

An even better reason to move this method to the `Module` class would be the current way that we use to construct the SQL for the queries. Right now this method is responsible for two different ways of building queries. How about if we define `get_min_max_object_id` in the `Module` class, and have its implementation without `DISTINCT`, and then override this method for the `Term_Relationships` module, and have the distinct implementation there?

What do you think?

Reply:
I love it

Reply:
It's not only called from the module class. The histogram method also uses it.

Reply:
But it can also use it from the sync module, no? I've pushed a commit to try that.